src/Http/Authentication.Core/test/AuthenticationPropertiesTests.cs

```diff
@@ -302,6 +303,41 @@ public void GetBool()
             Assert.Equal("BAR", props.Items["foo"]);
         }
 
+        [Fact]
+        public void Roundtrip_Serializes_With_SystemTextJson()
+        {
+            var props = new AuthenticationProperties()
+            {
+                AllowRefresh = true,
+                ExpiresUtc = new DateTimeOffset(2021, 03, 28, 13, 47, 00, TimeSpan.Zero),
+                IssuedUtc = new DateTimeOffset(2021, 03, 28, 12, 47, 00, TimeSpan.Zero),
+                IsPersistent = true,
+                RedirectUri = "/foo/bar"
+            };
+
+            props.Items.Add("foo", "bar");
+
+            props.Parameters.Add("baz", "quux");
+
+            var json = JsonSerializer.Serialize(props);
+            var deserialized = JsonSerializer.Deserialize<AuthenticationProperties>(json);
+
+            Assert.NotNull(deserialized);
+
+            Assert.Equal(props.AllowRefresh, deserialized!.AllowRefresh);
+            Assert.Equal(props.ExpiresUtc, deserialized.ExpiresUtc);
+            Assert.Equal(props.IssuedUtc, deserialized.IssuedUtc);
+            Assert.Equal(props.IsPersistent, deserialized.IsPersistent);
+            Assert.Equal(props.RedirectUri, deserialized.RedirectUri);
+
+            Assert.NotNull(deserialized.Items);
+            Assert.True(deserialized.Items.ContainsKey("foo"));
+            Assert.Equal(props.Items["foo"], deserialized.Items["foo"]);
+
+            Assert.NotNull(deserialized.Parameters);
```

Comment:
Maybe worth a comment saying something like // ensure parameters are not round tripped for extra clarity